lib/babe/babe.go

```diff
@@ -110,12 +111,16 @@ func NewSession(cfg *SessionConfig) (*Session, error) {
 		authorityData:    cfg.AuthData,
 		epochThreshold:   cfg.EpochThreshold,
 		startSlot:        cfg.StartSlot,
-		done:             cfg.Done,
+		epochDone:        cfg.EpochDone,
 		kill:             cfg.Kill,
-		closed:           false,
 		syncLock:         cfg.SyncLock,
 	}
 
+	canLock := atomic.CompareAndSwapUint32(&babeSession.closed, 0, 1)
+	if !canLock {
+		panic("[core] Error when trying to change Service status from stopped to started.")
```

Comment:
return error

Reply:
also, you are changing `closed` to 1 which doesn't really make sense here. change the variable name to `stopped` ?

also, shouldn't this be moved to `babe.Start` instead of `NewSession`? since it's not actually being started here

Reply:
I updated this to started, please re-check. Changing the lock to another method is another behavior change that will increase the scope. If this is required, we should do it as part of the same request for the core/service.go.